JuliaDiff/DifferentiationInterface.jl
# Add FastDifferentiation, simplify extras dispatch once more

**Core**
- Every operator has the default `extras=prepare_operator(backend, f, x, [y])`
- Add `SymbolicMode` and switch some trait dispatches to `AbstractMode`
- For second order, a single backend defaults to `SecondOrder(backend, backend)`
- Fallback hessian-vector product now exists for every `SecondOrder`

**Test**

- Remove tests with `extras=nothing` (divide number by 2)
- Remove Aqua ambiguity check (was detecting something that will never happen). Still useful to run it once in a while
- Add first reverse-over-forward hessian test

**Extensions**

- No need to specify `extras=nothing` in the extensions anymore, or do explicitly do preparation
- Add FastDifferentiation.jl symbolic support
- Fix same-type `Duplicated` for reverse allocating Enzyme

**Docs**

- Warning about uncertain second order backend combinations
